pkg/resource/notebook_instance_lifecycle_config/manager_factory.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package notebook_instance_lifecycle_config
+
+import (
+	"fmt"
+	"sync"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcfg "github.com/aws-controllers-k8s/runtime/pkg/config"
+	ackmetrics "github.com/aws-controllers-k8s/runtime/pkg/metrics"
+	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/go-logr/logr"
+
+	svcresource "github.com/aws-controllers-k8s/sagemaker-controller/pkg/resource"
+)
+
+// resourceManagerFactory produces resourceManager objects. It implements the
+// `types.AWSResourceManagerFactory` interface.
+type resourceManagerFactory struct {
+	sync.RWMutex
+	// rmCache contains resource managers for a particular AWS account ID
+	rmCache map[string]*resourceManager
+}
+
+// ResourcePrototype returns an AWSResource that resource managers produced by
+// this factory will handle
+func (f *resourceManagerFactory) ResourceDescriptor() acktypes.AWSResourceDescriptor {
+	return &resourceDescriptor{}
+}
+
+// ManagerFor returns a resource manager object that can manage resources for a
+// supplied AWS account
+func (f *resourceManagerFactory) ManagerFor(
+	cfg ackcfg.Config,
+	log logr.Logger,
+	metrics *ackmetrics.Metrics,
+	rr acktypes.Reconciler,
+	sess *session.Session,
+	id ackv1alpha1.AWSAccountID,
+	region ackv1alpha1.AWSRegion,
+) (acktypes.AWSResourceManager, error) {
+	rmId := fmt.Sprintf("%s/%s", id, region)
+	f.RLock()
+	rm, found := f.rmCache[rmId]
+	f.RUnlock()
+
+	if found {
+		return rm, nil
+	}
+
+	f.Lock()
+	defer f.Unlock()
+
+	rm, err := newResourceManager(cfg, log, metrics, rr, sess, id, region)
+	if err != nil {
+		return nil, err
+	}
+	f.rmCache[rmId] = rm
+	return rm, nil
+}
+
+// IsAdoptable returns true if the resource is able to be adopted
+func (f *resourceManagerFactory) IsAdoptable() bool {
```

Comment:
Have you tested/verified that it works properly.

Reply:
Yea adopted resources seem to work just fine.